src/stores/ExploreDataStore.ts

```diff
@@ -0,0 +1,130 @@
+import { makeAutoObservable } from "mobx";
+import { ConstraintInstance, TypeFormatTuple, WorkflowConfig, WorkflowSolution } from "./WorkflowTypes";
+import { makePersistable } from "mobx-persist-store";
+
+const emptyWorkflowConfig = () => {
+  return {
+    domain: undefined,
+    inputs: [ [{id:"",label:""}, {id:"",label:""}] as TypeFormatTuple ],
+    outputs: [ [{id:"",label:""}, {id:"",label:""}] as TypeFormatTuple],
+    constraints: [ {constraint: {id:"",label:""}} as ConstraintInstance ],
+    minSteps: 3,
+    maxSteps: 4,
+    timeout: 120,
+    solutionCount: 10,
+    run_id: ""
+  }
+}
+
+export class ExploreDataStore {
+
+  workflowConfig: WorkflowConfig = emptyWorkflowConfig();
+  workflowSolutions: WorkflowSolution[] = [];
+  isGenerating: boolean = false;
+  generationError: string = "";
+
+  constructor() {
+    makeAutoObservable(this, {}, { deep: true });
+    makePersistable(this, { 
+      name: "ExploreDataStore",
+      properties: ["workflowConfig", "workflowSolutions"],
+      storage: window.localStorage
+    });
+  }
+
+  inputsOutputsToJSON(values: TypeFormatTuple[], dataRoot: string, formatRoot: string) {
+    return values.filter(value => value[0] !== undefined && value[1] !== undefined && value[0]!.id !== "" && value[1]!.id !== "")
+      .map((value) => {
+        return {
+          [dataRoot]: [value[0]!.id.replace("http://edamontology.org/", "")],
+          [formatRoot]: [value[1]!.id.replace("http://edamontology.org/", "")]
+        };
+    });
+  }
+
+  configToJSON(config: WorkflowConfig): any {
+    const dataRoot: string = "data_0006";
+    const formatRoot: string = "format_1915";
+    const toolsRoot: string = "operation_0004";
+
+    const inputs = this.inputsOutputsToJSON(config.inputs, dataRoot, formatRoot);
+    const outputs = this.inputsOutputsToJSON(config.outputs, dataRoot, formatRoot);
+    
+    //TODO: figure out how much of this to hardcode
+    const obj: any = {
```

Comment:
Is this used in the code? If it is hardcoded, we could discuss how to change that

Reply:
I think at this point it was still unclear to me that the config to submit to APE is actually the domain config with some parameters filled in. I was probably confused because the config_path is also a parameter to the run_synthesis call (right?)

Reply:
Perhaps it would be good to fix these things before merging

Reply:
run_config only requires configJson (the content of the domain config + run parameters). This is something that we should discuss as this could be further improved (it should be sufficient to reference a domain that you are using, excluding the domain parameters from this configJson). This will be possible once REST APE supports local storage of the domains for faster processing

Reply:
So you mean that in the end, we don't have to retrieve the config from a URL, but could access it either through the API or APE?

Reply:
Maybe we should discuss this in person, we are flexible there. I was mostly referring to the fact that we could provide a simple "domain_init" call that parses and cashes a domain from a given URL, and every consecutive call that targets that domain (e.g., get tools/constraints, run synthesis) would reference the domain (e.g., domain ID) instead of providing the domain configuration again.

Reply:
We will leave this as a separate issue #10